aerc: integrate signature options

lukasngl opened the pull request:
### Description

Integrates the options defined in `accounts.email.accounts.<name>.signature`.

The options are Integrated as following:

1. If `signature.showSignature` is `none` don't generate signature config.
   (The `attach` variant does not have a corresponding option in aerc,
   so its treated like `append`)
2. If `signature.command` is not `null`, `signature-cmd` is set to a script, that:
   1. Prints the `signature.delimiter`
   2. Executes `signature.command`
3. Else `signature-file` is set to a file, where the content is the concatenation
    of `signature.delimiter` and `signature.text`.

*Note: This is based on #3150, as the upstream module is utterly broken.*

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}
    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

rycee commented:
Thanks for the contribution! Unfortunately the size will make it very difficult to review in a reasonable way as a whole. I would suggest creating separate PRs for each commit, rather than moving this one to ready for review.

The "add space after definitions" commit should either be first (and merged before opening the other PRs) or last, otherwise there will definitely be various conflicts.

lukasngl commented:
ups, sorry lost track of it. Have now rebased it on master.

lukasngl commented:
@teto would you mind taking a look at it?

lukasngl commented:
Hi, don't know why github marked this as having merge conflicts,
but anyways, I have now rebased on master.

I also figured out how to add test for the signature files,
and removed some trailing emtpy lines from the generated scripts.

I am not 100% sure on the format of the signature,
this PR uses `signature.delimiter` followed by `signature.text` or `signature.command`.

From what I have seen, neomutt and alot seem to ignore the `signature.delimiter`
and `signature.command` option and only use the `signature.text`.
Himalaya on the other hand uses the `signature.delimiter` but not the `signature.command`.

@mdietrich16 any opinions on this?

lukasngl commented:
If the `showSignature` option is set to `attach`, it is treated like `append` and a warning is shown.

This could be annoying, should we instead silently treat is like `attach` (as himalaya does) or only add the signature if the option is `append`?

lukasngl commented on modules/programs/aerc/accounts.nix:
```diff
@@ -288,11 +290,40 @@ in
+        if account.signature.showSignature == "append" then
+          if account.signature.command != null then
+            {
+              signature-cmd = writeShellScript "aerc-signature.sh" (
```
Hi, thanks for the review, but I'm not 100% sure what you mean exactly. According to [aerc-accounts(5)] *signature-cmd* is just the argument to "sh - c" and aerc doesn't have any notion of a *signature-delimiter* itself.

Do you mean, that we should set *signature-cmd* to "aerc-signature.sh ${delimiter}"?

I haven't seen that in the other mail modules. The idea was to introduce a similar behaviour as other mail clients so the config is portable (I.e. insert the delimiter before the signature). Although I don't know how other mail clients handle it as I personally haven't used signatures yet, so feedback is appreciated. 

[aerc-accounts(5)]: https://git.sr.ht/~rjarry/aerc/tree/master/item/doc/aerc-accounts.5.scd#L294

lukasngl commented:
hi, i just rebased on master again and requested reviews.

In the meantime you could test drive it as follows and maybe report back:

<details>
<summary>
As a flake user:
</summary>

```nix
  {
    inputs = {
      home-manager.url = "github:nix-community/home-manager";
      # Add this fork as an input
      home-manager-aerc-signature.url = "github:lukasngl/home-manager/aerc-signature";
    };

    outputs = { home-manager, home-manager-aerc-signature, ... }: {
      homeConfigurations."user" = home-manager.lib.homeManagerConfiguration {
        # ...
        modules = [
          {
            # Disable the upstrean aerc module
            disabledModules = [ "programs/aerc" ];
            # ... and replace it with the fork
            imports = [
              "${home-manager-aerc-signature}/modules/programs/aerc"
            ];
          }
          # ... rest of config
        ];
      };
    };
  }

```

</details>

<details>
<summary>
Educated guess for non-flake users:
</summary>

```nix
  { pkgs, ... }:

  let
    aerc-signature = pkgs.fetchFromGitHub {
      repo = "home-manager";
      owner = "lukasngl";
      ref = "aerc-signature";
      sha256 = ""; # Insert correct has here
    };
  in {
    disabledModules = [ "programs/aerc" ];
    imports = [
      "${aerc-signature}/modules/programs/aerc"
    ];

    # ... rest of config
  }

```

</details>
